File: cpp_tetris/src/Position.cpp
#include "tetris/Position.h"

namespace tetris {

Position::Position(int x, int y) : x_(x), y_(y) {}

int Position::getX() const {
    return x_;
}

int Position::getY() const {
    return y_;
}

void Position::setX(int x) {
    x_ = x;
}

void Position::setY(int y) {
    y_ = y;
}

Position Position::add(const Position& other) const {
    return Position(x_ + other.x_, y_ + other.y_);
}

bool Position::operator==(const Position& other) const {
    return x_ == other.x_ && y_ == other.y_;
}

bool Position::operator!=(const Position& other) const {
    return !(*this == other);
}

} // namespace tetris

File: js-src/position.js
/**
 * Represents a 2D position with x and y coordinates.
 * Converted from Java Position class.
 */
export class Position {
  /**
   * Creates a new Position instance.
   * @param {number} x - The x coordinate
   * @param {number} y - The y coordinate
   */
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }

  /**
   * Gets the x coordinate.
   * @returns {number} The x coordinate
   */
  getX() {
    return this.x;
  }

  /**
   * Gets the y coordinate.
   * @returns {number} The y coordinate
   */
  getY() {
    return this.y;
  }

  /**
   * Sets the x coordinate.
   * @param {number} x - The new x coordinate
   */
  setX(x) {
    this.x = x;
  }

  /**
   * Sets the y coordinate.
   * @param {number} y - The new y coordinate
   */
  setY(y) {
    this.y = y;
  }

  /**
   * Adds another position to this position and returns a new Position.
   * @param {Position} other - The position to add
   * @returns {Position} A new Position with the sum of coordinates
   */
  add(other) {
    return new Position(this.x + other.x, this.y + other.y);
  }
}